tests/std/tests/P0226R1_math_special_functions/test.cpp

```diff
@@ -0,0 +1,540 @@
+        assert(isclose(riemann_zetal(-1.0L), -0.0833'333'333'333'333'333L));
+    }
+    {
+        assert(isclose(riemann_zeta(0.0), -0.5));
+        assert(isclose(riemann_zeta(0.0f), -0.5f));
+        assert(isclose(riemann_zeta(0.0L), -0.5L));
+        assert(isclose(riemann_zetaf(0.0f), -0.5f));
+        assert(isclose(riemann_zetal(0.0L), -0.5L));
+    }
+}
+
+void test_sph_bessel() {
+    // sph_bessel(x, y) = sqrt[pi/(2 * x)] BesselJ[y + 1/2, x]
+    {
+        assert(isnan(sph_bessel(1, nand_v)));
+        assert(isnan(sph_bessel(1, nanf_v)));
+        assert(isnan(sph_bessel(1, nanl_v)));
+        assert(isnan(sph_besself(1, nanf_v)));
+        assert(isnan(sph_bessell(1, nanl_v)));
+    }
+    {
+        assert(isclose(sph_bessel(1, 1.2345), 0.352'106'432'958'031'885));
+        assert(isclose(sph_bessel(1, 1.2345f), 0.352'106'432'958'031'885f));
+        assert(isclose(sph_bessel(1, 1.2345L), 0.352'106'432'958'031'885L));
+        assert(isclose(sph_bessel(1, 1.2345f), 0.352'106'432'958'031'885f));
+        assert(isclose(sph_bessel(1, 1.2345L), 0.352'106'432'958'031'885L));
+    }
+}
+
+void test_sph_legendre() {
+    // sph_legendre(x, y, z)
+    // = (-1)^y ([2 * x + 1]/4pi * [x - y]!/[x + y]!)^(1 / 2) LegendreP[x, y, cos(z)]
+    {
+        assert(isnan(sph_legendre(1, 1, nand_v)));
+        assert(isnan(sph_legendre(1, 1, nanf_v)));
+        assert(isnan(sph_legendre(1, 1, nanl_v)));
+        assert(isnan(sph_legendref(1, 1, nanf_v)));
+        assert(isnan(sph_legendrel(1, 1, nanl_v)));
+    }
+    {
+        assert(isclose(sph_legendre(3, 0, 1.2345), -0.302'386'890'645'537'993, 2));
+        assert(isclose(sph_legendre(3, 0, 1.2345f), -0.302'386'890'645'537'993f));
+        assert(isclose(sph_legendre(3, 0, 1.2345L), -0.302'386'890'645'537'993L, 2));
+        assert(isclose(sph_legendre(3, 0, 1.2345f), -0.302'386'890'645'537'993f));
+        assert(isclose(sph_legendre(3, 0, 1.2345L), -0.302'386'890'645'537'993L, 2));
+    }
+}
+
+void test_sph_neumann() {
+    // sph_neumann(x, y) = sqrt[pi/2y] BesselY[x + 1/2, y]
+    {
+        assert(isnan(sph_neumann(1, nand_v)));
+        assert(isnan(sph_neumann(1, nanf_v)));
+        assert(isnan(sph_neumann(1, nanl_v)));
+        assert(isnan(sph_neumannf(1, nanf_v)));
+        assert(isnan(sph_neumannl(1, nanl_v)));
+    }
+    {
+        assert(isclose(sph_neumann(1, 1.2345), -0.981'200'877'819'308'099, 2));
+        assert(isclose(sph_neumann(1, 1.2345f), -0.981'200'877'819'308'099f));
+        assert(isclose(sph_neumann(1, 1.2345L), -0.981'200'877'819'308'099L, 2));
+        assert(isclose(sph_neumann(1, 1.2345f), -0.981'200'877'819'308'099f));
+        assert(isclose(sph_neumann(1, 1.2345L), -0.981'200'877'819'308'099L, 2));
+    }
+}
+
+// Also test GH-3076 <cmath>: Invalid output for incomplete elliptic integral of the second kind with k = 1
+void test_gh_3076() {
+    assert(isclose(ellint_2(1, 6.2831853071795862), 4.0));

```

Comment:
This regression test unsurprisingly fails when run against the "old" boost::math version built into the special math satellite in the locked redist. Since our internal testing always (currently) uses the locked redist, it will fail in MS-internal testing when linking against the STL DLLs in the redist. Luckily, we can guard so this is case is not exercised until the next redist unlock.